### ISSUE TITLE
Proposed updates to documentation for `reader_name` argument of nvidia.dali.plugin.*.DALI*Iterator

### ISSUE THREAD

**Issue by matthew-frank** [NVIDIA/DALI#4177]
The `reader_name` argument of `nvidia.dali.plugin.*.DALI*Iterator()` has been difficult for us to understand.  I'd like to propose a rewording of the documentation, but want to check that what I'm proposing is actually correct, and ask some questions.

The current documentation is:
> **reader_name** (*str, default = None*) - Name of the reader which will be queried to the shard size, number of shards and all other properties necessary to count properly the number of relevant and padded samples that iterator needs to deal with. It automatically sets last_batch_policy to PARTIAL when the FILL is used, and last_batch_padded accordingly to match the reader’s configuration

1. I believe the final sentence is incorrect and should be deleted?  We have experimented with setting `last_batch_policy` to `LastBatchPolicy.FILL` and it seems to work as described in the examples (i.e., it works like "FILL", rather than being overridden to "PARTIAL".)
2. Rather: I think that the semantics of using this option is that using `reader_name` is mutually exclusive with using the `last_batch_padded` argument, and instead, if the `reader_name` argument is set, the `last_batch_padded` argument is set to the value of the `pad_last_batch` option of the reader?
3. The documentation says that the `reader_name` argument is the "name of the reader which will be queried" but doesn't explain what object will be queried to find the reader.  Is it the list of pipelines given by the `pipelines` argument?  What are the semantics if the reader is not found in any of the pipelines from the `pipelines` argument?  What are the semantics if the reader is found in more than one of the pipelines form the `pipelines` argument?
4. I can't find the documentation for how to set the name of the Reader in a Pipeline.  I see that in the [MXNet with DALI - ResNet 50 example](https://docs.nvidia.com/deeplearning/dali/user-guide/docs/examples/use_cases/mxnet/mxnet-resnet50.html), the `fn.readers.mxnet` constructor apparently takes an argument `name`, but this argument isn't documented in the documentation for [`fn.readers.mxnet()`](https://docs.nvidia.com/deeplearning/dali/user-guide/docs/operations/nvidia.dali.fn.readers.mxnet.html#nvidia.dali.fn.readers.mxnet), (nor in any of the documentation for any of the other readers), and the semantics of naming readers in Pipelines doesn't seem to be documented.  The only place the reader name is mentioned in the documentation is in the documentation for the `reader_meta` argument to [Pipeline](https://docs.nvidia.com/deeplearning/dali/user-guide/docs/pipeline.html).
5. Finally, I also wonder if the `size` argument to the `DALI*Iterator()` is correctly documented.  It says that if the `size` is set to the default (-1), "The options last_batch_policy and last_batch_padded don’t work in such case." but then it also says setting size=-1 is mutually exclusive with the reader_name argument.

**Comment by awolant** [comment]
Thanks for reporting the issue. Let me check this and get back to you. We will work something out to make this more understandable.

**Comment by Bill-Gets** [comment]
I can't agree more with you. You really speak to my heart. Several months passed and sadly the documentation is still the same as here at present.

To be honest, there're millions of doubts when I read the documentation. I have no idea about what the `size` means. Does it 
stand for batch size? And what is the epoch size in the documentation? What should I fill in the `reader_name`? What does `reader_name = "Readers"` means? Where does the "readers" come from? I feel so helpless...

**Comment by dyhan316** [comment]
I also agree. While the "Getting started tutorial" helped a lot, I believe that it should have an explanation/example of using "reader_name".
